wandb/sdk/internal/file_stream.py

```diff
@@ -238,47 +239,37 @@ def _read_queue(self):
             self._queue, self.MAX_ITEMS_PER_PUSH, self.rate_limit_seconds()
         )
 
-    def _thread_body(self):
-        posted_data_time = time.time()
-        posted_anything_time = time.time()
-        ready_chunks = []
-        uploaded = set()
-        finished = None
-        while finished is None:
+    def process_queue(self):
+        while True:
             items = self._read_queue()
+            if len(items) == 0:
+                break
+
             for item in items:
                 if isinstance(item, self.Finish):
-                    finished = item
-                elif isinstance(item, self.Preempting):
-                    request_with_retry(
-                        self._client.post,
-                        self._endpoint,
-                        json={
-                            "complete": False,
-                            "preempting": True,
-                            "dropped": self._dropped_chunks,
-                            "uploaded": list(uploaded),
-                        },
-                    )
-                    uploaded = set()
+                    self.finished = item
                 elif isinstance(item, self.PushSuccess):
-                    uploaded.add(item.save_name)
+                    self.uploaded.add(item.save_name)
                 else:
                     # item is Chunk
-                    ready_chunks.append(item)
+                    self.ready_chunks.append(item)
 
             cur_time = time.time()
 
-            if ready_chunks and (
-                finished or cur_time - posted_data_time > self.rate_limit_seconds()
+            if self.ready_chunks and (
+                self.finished
+                or cur_time - self.posted_data_time > self.rate_limit_seconds()
             ):
-                posted_data_time = cur_time
-                posted_anything_time = cur_time
-                self._send(ready_chunks)
-                ready_chunks = []
-
-            if cur_time - posted_anything_time > self.heartbeat_seconds:
-                posted_anything_time = cur_time
+                self.posted_data_time = cur_time
+                self.posted_anything_time = cur_time
+                self._send(self.ready_chunks)
+                self.ready_chunks = []
+
+            if (
+                self.posted_anything_time
+                and self.posted_anything_time > self.heartbeat_seconds
```

Comment:
Accidentally dropped the `cur_time -`?
```suggestion
                and cur_time - self.posted_anything_time > self.heartbeat_seconds
```